src/test/scala/is/hail/methods/LinearRegressionSuite.scala

```diff
@@ -152,7 +152,95 @@ class LinearRegressionSuite extends SparkSuite {
+    x = c(0.009900990296049406, 0.7450495050747477, 0.9900990100009803, 0.9900990100009803, 0.9900990100009803, 0.7450495050747477)
+    */
+
+    assertDouble(qBeta, v3, 1.09536219)
+    assertDouble(qSe, v3, 0.6901002)
+    assertDouble(qTstat, v3, 1.5872510)
+    assertDouble(qPval, v3, 0.2533675)
+
+    assertEmpty(qBeta, v6)
+  }
+
+  @Test def testWithTwoCovDosage() {
+    // .gen and .sample files created from regressionLinear.vcf
+    // dosages are derived from PLs so results should agree with testWithTwoCovPhred
+    val vds = hc.importGen("src/test/resources/regressionLinear.gen", "src/test/resources/regressionLinear.sample")
+      .filterMulti()
+      .annotateSamplesTable("src/test/resources/regressionLinear.cov",
+        "Sample",
+        root = Some("sa.cov"),
+        config = TextTableConfiguration(types = Map("Cov1" -> TDouble, "Cov2" -> TDouble)))
+      .annotateSamplesTable("src/test/resources/regressionLinear.pheno",
+        "Sample",
+        root = Some("sa.pheno"),
+        config = TextTableConfiguration(types = Map("Pheno" -> TDouble), missing = "0"))
+      .linreg("sa.pheno.Pheno", Array("sa.cov.Cov1", "sa.cov.Cov2 + 1 - 1"), "va.linreg", useDosages = true, 1, 0.0)
+
+    val v1 = Variant("1", 1, "C", "T")   // x = (0, 1, 0, 0, 0, 1)
+    val v2 = Variant("1", 2, "C", "T")   // x = (., 2, ., 2, 0, 0)
+    val v3 = Variant("1", 3, "C", "T")   // x = (0, ., 1, 1, 1, .)
+    val v6 = Variant("1", 6, "C", "T")   // x = (0, 0, 0, 0, 0, 0)
+    val v7 = Variant("1", 7, "C", "T")   // x = (1, 1, 1, 1, 1, 1)
+    val v8 = Variant("1", 8, "C", "T")   // x = (2, 2, 2, 2, 2, 2)
+    val v9 = Variant("1", 9, "C", "T")   // x = (., 1, 1, 1, 1, 1)
+    val v10 = Variant("1", 10, "C", "T") // x = (., 2, 2, 2, 2, 2)
+
+    val qBeta = vds.queryVA("va.linreg.beta")._2
+    val qSe = vds.queryVA("va.linreg.se")._2
+    val qTstat = vds.queryVA("va.linreg.tstat")._2
+    val qPval = vds.queryVA("va.linreg.pval")._2
+
+    val annotationMap = vds.variantsAndAnnotations
+      .collect()
+      .toMap
 
+    def assertInt(q: Querier, v: Variant, value: Int) =
+      assert(D_==(q(annotationMap(v)).asInstanceOf[Int], value))
+
+    def assertDouble(q: Querier, v: Variant, value: Double) =
+      assert(D_==(q(annotationMap(v)).asInstanceOf[Double], value, tolerance = 1e-4))
+
+    def assertEmpty(q: Querier, v: Variant) =
+      assert(q(annotationMap(v)) == null)
+
+    val g0 = Genotype.apply(gt = Some(0), px=Some(Array(32648, 20, 100)), isDosage = true)
+    val g1 = Genotype.apply(gt = Some(1), px=Some(Array(20, 32648, 100)), isDosage = true)
+    val g2 = Genotype.apply(gt = Some(2), px=Some(Array(20, 100, 32648)), isDosage = true)
+
+    assert(D_==(g0.unboxedBiallelicDosageGenotype, (20 + 2 * 100) / 32768d))
+    assert(D_==(g1.unboxedBiallelicDosageGenotype, (32648 + 2 * 100) / 32768d))
+    assert(D_==(g2.unboxedBiallelicDosageGenotype, (100 + 2 * 32648) / 32768d))
```

Comment:
These six lines also seem independent of this suite. Put them in `GenotypeSuite`.